.circleci/config.yml

```diff
@@ -0,0 +1,515 @@
+version: 2.1
+
+references:
+  ci_support_image: &ci_support_image
+                       broadinstitute/study-server-build:2020-04-10G
+  root_path: &root_path
+      /home/circleci
+  repo_path: &repo_path
+              /home/circleci/repo
+  m2_cache_key: &m2_cache_key
+                  pom.xml1-{{ checksum "pom.xml" }}
+  m2_cache_restore_key: &m2_cache_keys
+    keys:
+      - *m2_cache_key
+      - pom.xml1-
+  m2_repo_path: &m2_repo_path
+             /home/circleci/.m2/repository
+  pepper_apis-path: &pepper_apis_path
+            /home/circleci/repo/pepper-apis
+  circleci_path: &circleci_path
+            /home/circleci/repo/.circleci
+  builds_bucket: &builds_bucket
+            ddp-build-artifacts
+  jar_build_bucket_dir: &jar_build_bucket_dir
+        /ddp-study-server-pepper-apis
+  housekeeping_test_db: &housekeeping_test_db
+      housekeepingdb
+  studyserver_test_db: &studyserver_test_db
+      studyservicedb
+
+
+executors:
+  build-executor:
+    docker:
+      - image: *ci_support_image
+  test-executor:
+    docker:
+      - image: *ci_support_image
+      # start pubsub emulator
+      - image: broadinstitute/study-server-build:pubsub-1
+      - image: circleci/mysql:5.7-ram
+        environment: &DB_VARS
+          MYSQL_ROOT_PASSWORD: rootpw
+          MYSQL_DATABASE: test_db
+          MYSQL_USER: user
+          MYSQL_PASSWORD: passw0rd
+    resource_class: medium+ #default medium not big enough for tests
+
+  deploy-executor:
+    docker: # todo: create new image endpoint, add this in admin tab for faster starts
+    - image: *ci_support_image
+
+commands:
+  compile:
+    description: Run maven compile
+    parameters:
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+      root_path:
+        type: string
+        default: *root_path
+    steps:
+      - restore_cache: *m2_cache_keys
+      - run:
+          name: Run maven
+          command: |
+            # Some images the $HOME directory is not writable. Let's be specific for .m2/repository location
+            mvn compile package -DskipTests -file pom.xml -Dmaven.repo.local=<< parameters.m2_repo >>
+            mvn package -DskipTests -f housekeeping-pom.xml -Dcheckstyle.skip -Dmaven.repo.local=<< parameters.m2_repo >>
+      - save_cache:
+          key: *m2_cache_key
+          paths:
+            - << parameters.root_path >>/.m2
+
+  setup-shared-env:
+    description: "Setup shared ENV vars used by multiple scripts"
+    steps:
+      - run:
+          name: Set environment to be used by build
+          command: |
+            echo "export VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_ROLE_SECRET_ID)" >> $BASH_ENV
+            echo "export SHORT_GIT_SHA=$(echo '<< pipeline.git.revision >>' | cut -c 1-7)" >> $BASH_ENV
+            source $BASH_ENV
```

Comment:
Where we generate the vault token. Uses CircleCi configuration. VAULT_ROLE_ID is injected by CircleCi